diff --git a/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.d.ts b/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.d.ts
index a68d623..eff11aa 100644
--- a/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.d.ts
+++ b/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.d.ts
@@ -10,7 +10,8 @@ declare const enum StreamingStatus {
  * @group Camera
  */
 export declare const enum VideoCodecType {
-    H264 = 0
+    H264 = 0,
+    H265 = 1
 }
 /**
  * @group Camera
@@ -99,6 +100,7 @@ export type VideoStreamingOptions = {
  * @group Camera
  */
 export interface H264CodecParameters {
+    type?: VideoCodecType;
     levels: H264Level[];
     profiles: H264Profile[];
 }
diff --git a/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.js b/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.js
index 1f64310..8120013 100644
--- a/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.js
+++ b/node_modules/@homebridge/hap-nodejs/dist/lib/camera/RTPStreamManagement.js
@@ -58,7 +58,7 @@ var VideoCodecType;
 (function (VideoCodecType) {
     VideoCodecType[VideoCodecType["H264"] = 0] = "H264";
     // while the namespace is already reserved for H265 it isn't currently supported.
-    // H265 = 0x01,
+    VideoCodecType[VideoCodecType["H265"] = 1] = "H265";
 })(VideoCodecType || (exports.VideoCodecType = VideoCodecType = {}));
 /**
  * @group Camera
@@ -882,7 +882,8 @@ class RTPStreamManagement {
                 tlv.encode(4 /* VideoCodecParametersTypes.CVO_ENABLED */, 1 /* VideoCodecCVO.SUPPORTED */, 5 /* VideoCodecParametersTypes.CVO_ID */, videoOptions.cvoId),
             ]);
         }
-        const videoStreamConfiguration = tlv.encode(1 /* VideoCodecConfigurationTypes.CODEC_TYPE */, 0 /* VideoCodecType.H264 */, 2 /* VideoCodecConfigurationTypes.CODEC_PARAMETERS */, codecParameters, 3 /* VideoCodecConfigurationTypes.ATTRIBUTES */, videoOptions.resolutions.map(resolution => {
+        const codecType = videoOptions.codec.type !== undefined ? videoOptions.codec.type : 0 /* VideoCodecType.H264 */;
+        const videoStreamConfiguration = tlv.encode(1 /* VideoCodecConfigurationTypes.CODEC_TYPE */, codecType, 2 /* VideoCodecConfigurationTypes.CODEC_PARAMETERS */, codecParameters, 3 /* VideoCodecConfigurationTypes.ATTRIBUTES */, videoOptions.resolutions.map(resolution => {
             if (resolution.length !== 3) {
                 throw new Error("Unexpected video resolution");
             }
